substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/io/Utf8.java

```diff
@@ -36,8 +36,22 @@ public final class Utf8 {
     private Utf8() {
     }
 
+    private static int utf8Size(char c) {
+        // Based On
+        // https://github.com/openjdk/jdk21u/blob/d6e8788ca888fac7f04751e72202f10fda8cad59/src/hotspot/share/utilities/utf8.cpp#L409-L418
```

Comment:
I think we should better use the following format (to reduce the chance of conflicts in the future).

```suggestion
    // @BasedOnJDKFile("https://github.com/openjdk/jdk21u/blob/d6e8788ca888fac7f04751e72202f10fda8cad59/src/hotspot/share/utilities/utf8.cpp#L409-L418")
    private static int utf8Size(char c) {
```

or completely remove the line.

@jerboaa WDYT?

Reply:
No strong feelings. On second thought, perhaps using tagged version links would be better (instead of Git hashes - permalinks)?

```
https://github.com/openjdk/jdk21u/blob/jdk-21.0.4%2B7/src/hotspot/share/utilities/utf8.cpp#L409-L418
```

Reply:
This gives better transparency when (which JDK 21 update release) a change got introduced.

Reply:
@luneo7 sorry for the back and forth in this PR, but can you please amend your commit with the following patch as I am not allowed to push on your branch? 

```diff
diff --git a/substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/io/Utf8.java b/substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/io/Utf8.java
index 31fe4defd81..b7204d2fa6c 100644
--- a/substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/io/Utf8.java
+++ b/substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/io/Utf8.java
@@ -38,7 +38,7 @@ public final class Utf8 {
 
     private static int utf8Size(char c) {
         // Based On
-        // https://github.com/openjdk/jdk21u/blob/d6e8788ca888fac7f04751e72202f10fda8cad59/src/hotspot/share/utilities/utf8.cpp#L409-L418
+        // https://github.com/openjdk/jdk21u/blob/jdk-21.0.6%2B7/src/hotspot/share/utilities/utf8.cpp#L409-L418
         if ((0x0001 <= c) && (c <= 0x007F)) {
             // ASCII character
             return 1;
@@ -66,7 +66,7 @@ public final class Utf8 {
      */
     public static int utf8Length(String s, int beginIndex, int endIndex) {
         // Based on
-        // https://github.com/openjdk/jdk21u/blob/d6e8788ca888fac7f04751e72202f10fda8cad59/src/hotspot/share/utilities/utf8.cpp#L433-L444
+        // https://github.com/openjdk/jdk21u/blob/jdk-21.0.6%2B7/src/hotspot/share/utilities/utf8.cpp#L433-L444
         if (beginIndex < 0 || endIndex > s.length() || beginIndex > endIndex) {
             throw new StringIndexOutOfBoundsException();
         }
diff --git a/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/util/Utf8.java b/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/util/Utf8.java
index 276af83c9d6..451c1ad5f2f 100644
--- a/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/util/Utf8.java
+++ b/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/util/Utf8.java
@@ -39,9 +39,8 @@ public final class Utf8 {
     private Utf8() {
     }
 
+    // @BasedOnJDKFile("https://github.com/openjdk/jdk21u/blob/jdk-21.0.6%2B7/src/hotspot/share/utilities/utf8.cpp#L409-L418)
     private static int utf8Size(char c) {
-        // Based On
-        // https://github.com/openjdk/jdk21u/blob/d6e8788ca888fac7f04751e72202f10fda8cad59/src/hotspot/share/utilities/utf8.cpp#L409-L418
         if ((0x0001 <= c) && (c <= 0x007F)) {
             // ASCII character
             return 1;
@@ -66,9 +65,8 @@ public final class Utf8 {
      * @return the length as {@code int} in bytes of the UTF8 representation of the string. Might
      *         return a truncated size if the value does not fit into {@code int} (see JDK-8328877).
      */
+    // @BasedOnJDKFile("https://github.com/openjdk/jdk21u/blob/jdk-21.0.6%2B7/src/hotspot/share/utilities/utf8.cpp#L433-L444")
     public static int utf8Length(String s, int beginIndex, int endIndex) {
-        // Based on
-        // https://github.com/openjdk/jdk21u/blob/d6e8788ca888fac7f04751e72202f10fda8cad59/src/hotspot/share/utilities/utf8.cpp#L433-L444
         if (beginIndex < 0 || endIndex > s.length() || beginIndex > endIndex) {
             throw new StringIndexOutOfBoundsException();
         }

```

Thanks